### PR TITLE
Program Manager with Upload and Downloads and banners for all of the …

### PR DESCRIPTION
**Pull request by reillz10**
…validations there in

### CODE REVIEW

File: src/upload/UploadPage.jsx [hunk 1]
```diff
@@ -23,16 +40,71 @@ class UploadPage extends React.Component {
...
+                onClick={() => this.handleDownloadProgramEnrollments(program.programKey)}
+              >Download Program Enrollments
+              </button>
+            </div>
             <div className="btn-group" role="group">
-              <button className="btn btn-outline-primary">Upload Program Enrollments</button>
-              <button className="btn btn-outline-primary">Upload Course Enrollments</button>
-              <button className="btn btn-outline-primary">Download Program Enrollments</button>
-              <button className="btn btn-outline-primary">Download Course Enrollments</button>
+              <button className="btn btn-outline-primary">
```
Comment (schenedx) [inline src/upload/UploadPage.jsx]: #NIT should we consider moving the upload button into it's own component file? The styles are repeated twice.

File: src/upload/sagas.js [hunk 2]
```diff
@@ -19,14 +37,159 @@ export function* handleFetchWritablePrograms() {
...
+    yield put(fetchJobsBegin());
+    const data = yield call(ApiService.getJobs);
+    for (let i = 0; i < data.length; i += 1) {
+      const job = data[i];
+      const jobNameInfo = parseRegistrarJobName(job.name);
+      if (jobNameInfo !== null) {
+        const bannerObj = {
+          id: jobNameInfo.programKey + Date.now(),
+          bannerType: 'info',
+          message: `An existing in-progress ${jobNameInfo.jobName} job has been found for this program`,
```
Comment (schenedx) [inline src/upload/sagas.js]: Should we use translation utility here so this line gets translated?

File: src/upload/sagas.js [hunk 3]
```diff
@@ -19,14 +37,159 @@ export function* handleFetchWritablePrograms() {
...
+export function* handleUploadEnrollments({ payload: { programKey, isCourses, file } }) {
+  try {
+    if (typeof file === 'undefined') {
+      return;
+    }
+    const data = yield call(ApiService.uploadEnrollments, programKey, isCourses, file);
+    const bannerObj = {
+      id: programKey + Date.now(),
+      bannerType: 'info',
+      message: 'Your enrollment file has been accepted and is being processed. Please wait.',
```
Comment (schenedx) [inline src/upload/sagas.js]: Same, we need to wrap this text with translation function

File: src/upload/sagas.js [hunk 4]
```diff
@@ -19,14 +37,159 @@ export function* handleFetchWritablePrograms() {
...
+  }
+}
+
+export function* handleDownloadEnrollments({ payload: { programKey, isCourses } }) {
+  try {
+    const data = yield call(ApiService.downloadEnrollments, programKey, isCourses);
+    const bannerObj = {
+      id: programKey + Date.now(),
+      bannerType: 'info',
+      message: 'Your enrollment csv is being built. Please wait.',
```
Comment (schenedx) [inline src/upload/sagas.js]: Translation?

File: src/upload/sagas.js [hunk 5]
```diff
@@ -19,14 +37,159 @@ export function* handleFetchWritablePrograms() {
...
+    if (responseData.state === 'Succeeded') {
+      const jobNameInfo = parseRegistrarJobName(responseData.name);
+      yield all([
+        put(removeBanner(programKey, bannerId)),
+        put(pollJobSuccess(
+          programKey,
+          {
+            id: programKey + Date.now(),
+            bannerType: 'success',
+            message: `Your ${jobNameInfo.jobName} job has has successfully completed.`,
```
Comment (schenedx) [inline src/upload/sagas.js]: Translation?

File: src/upload/UploadPage.jsx (hunk 1 above)
Comment (reillz10) [reply src/upload/UploadPage.jsx]: Eventually, I think there is room to refactor this but would be hesitant to do so as of now without knowing how Program Manager may evolve in future iterations

**Pull request merged**